test/rtos_drivers/hil_add/src/individual_tests/uart/local/tx_write_rx_read_test.c

```diff
@@ -0,0 +1,155 @@
+// Copyright 22022 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+/* System headers */
+#include <platform.h>
+#include <xs1.h>
+#include <string.h>
+
+/* FreeRTOS headers */
+#include "FreeRTOS.h"
+
+/* Library headers */
+#include "rtos_uart_tx.h"
+#include "rtos_uart_rx.h"
+
+/* App headers */
+#include "app_conf.h"
+#include "individual_tests/uart/uart_test.h"
+
+#if ON_TILE(UART_RX_TILE)
+static volatile int error_cb_occurred = 0;
+
+UART_RX_STARTED_ATTR
+void rx_started(rtos_uart_rx_t *ctx){
+    uart_printf("rtos_uart_rx_started_cb");
+}
+
+UART_RX_ERROR_ATTR
+void rx_error(rtos_uart_rx_t *ctx, uint8_t err_flags){
+
+    uart_printf("rtos_uart_rx_error:");
+    if(err_flags & UR_START_BIT_ERR_CB_FLAG){
+        uart_printf("UART_START_BIT_ERROR");
+    }
+    if(err_flags & UR_PARITY_ERR_CB_FLAG){
+        uart_printf("UART_PARITY_ERROR");
+    }
+    if(err_flags & UR_FRAMING_ERR_CB_FLAG){
+        uart_printf("UART_FRAMING_ERROR");
+    }
+    if(err_flags & UR_OVERRUN_ERR_CB_FLAG){
+        uart_printf("UR_OVERRUN_ERR_CB_CODE");
+    }
+
+    if(err_flags & ~ RX_ERROR_FLAGS){
+        uart_printf("UNKNOWN ERROR FLAG SET: 0x%x (THIS SHOULD NEVER HAPPEN)", err_flags);
+    }
+
+    error_cb_occurred = 1;
+}
+
+UART_RX_COMPLETE_ATTR
+void rx_complete(rtos_uart_rx_t *ctx){
+    // uart_printf("UART_RX_COMPLETE\n");
+    // The backpressure from this causes a test fail - WHY?
+}
+#endif
+
+static const char* test_name = "uart_loopback_test";
+
+UART_MAIN_TEST_ATTR
+static int main_test(uart_test_ctx_t *ctx)
+{
+    int retval = 0;
+    const unsigned num_packets = 100;
+    const uint8_t tx_buff[] = {0xed, 0x00, 0x77, 0xed, 0x00, 0x77, 0xed, 0x00, 0x55, 0x55, 0xff, 0x55,
+                               0xed, 0x00, 0x77, 0xed, 0x00, 0x77, 0xed, 0x00, 0x55, 0x55, 0xff, 0x55,
+                               0xed, 0x00, 0x77, 0xed, 0x00, 0x77, 0xed, 0x00, 0x55, 0x55, 0xff, 0x55,
+                               0xed, 0x00, 0x77, 0xed, 0x00, 0x77, 0xed, 0x00, 0x55, 0x55, 0xff, 0x55,
+                               0xed, 0x00, 0x77, 0xed, 0x00, 0x77, 0xed, 0x00, 0x55, 0x55, 0xff, 0x55 };//6 x 12 = 72B
+
+#if ON_TILE(UART_TX_TILE)
+    for(unsigned t = 0; t < num_packets; t++)
+    {
+        rtos_uart_tx_write(ctx->rtos_uart_tx_ctx, tx_buff, sizeof(tx_buff));
```

Comment:
Looks good. Wish I had merged https://github.com/xmos/xcore_sdk/pull/486 before though as I will now need to update that. My bad..